src/Development/IDE/Core/Shake.hs

```diff
@@ -258,7 +258,22 @@ setValues :: IdeRule k v
           -> IO ()
 setValues state key file val = modifyVar_ state $ \vals -> do
     -- Force to make sure the old HashMap is not retained
-    evaluate $ HMap.insert (file, Key key) (fmap toDyn val) vals
+    let v = fmap toDyn val
```

Comment:
Do we need to force `val` as well? `Dynamic` is not strict in the value I believe.

Reply:
Indeed, Dynamic is not strict in the value. But we don't need to force it, and forcing it doesn't eliminate the space leak, and we don't subsequently force it in the lookup anyway. I tried these things, including using seqValue, but they didn't have any impact.